# Data to Analyze profit from Apps 
our main objective is to guess the profit we can earn from the which type of app.
Our goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more users.

In [1]:
from csv import reader

### The Google Play data set ###
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### The App Store data set ###
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]





    

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line between rows
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

print(ios_header)
print('\n')
explore_data(android, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


In [3]:
duplicate_apps=[]
unique_apps=[]

for row in android:
    name=row[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
print(len(unique_apps))
print(len(duplicate_apps))


9660
1181


In [4]:

print(android_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [5]:
reviews_max = {}
clean_reviews=[]
for app in android:
    name = app[0]
    raw_reviews = app[3]

    try:
        # Clean and convert review count to float
        if 'M' in raw_reviews:
            n_reviews = float(raw_reviews.replace('M', '')) * 1_000_000
           
        elif 'K' in raw_reviews:
            n_reviews = float(raw_reviews.replace('K', '')) * 1_000
            
        else:
            n_reviews = float(raw_reviews.replace(',', ''))  # In case it has commas
            
        # Store max review count
        if name in reviews_max  and reviews_max[name] < n_reviews:
            
                reviews_max[name] = n_reviews
                
        elif name not in reviews_max:
            reviews_max[name] = n_reviews
            

    except ValueError:
        print(f"Skipping app: {name}, invalid review format: {raw_reviews}")


In [6]:
count=0
for key,value in reviews_max.items():
    count=count+1
    print(f'Reviews of {key} is {value}')
   

Reviews of Photo Editor & Candy Camera & Grid & ScrapBook is 159.0
Reviews of Coloring book moana is 974.0
Reviews of U Launcher Lite – FREE Live Cool Themes, Hide Apps is 87510.0
Reviews of Sketch - Draw & Paint is 215644.0
Reviews of Pixel Draw - Number Art Coloring Book is 967.0
Reviews of Paper flowers instructions is 167.0
Reviews of Smoke Effect Photo Maker - Smoke Editor is 178.0
Reviews of Infinite Painter is 36815.0
Reviews of Garden Coloring Book is 13791.0
Reviews of Kids Paint Free - Drawing Fun is 121.0
Reviews of Text on Photo - Fonteee is 13880.0
Reviews of Name Art Photo Editor - Focus n Filters is 8788.0
Reviews of Tattoo Name On My Photo Editor is 44829.0
Reviews of Mandala Coloring Book is 4326.0
Reviews of 3D Color Pixel by Number - Sandbox Art Coloring is 1518.0
Reviews of Learn To Draw Kawaii Characters is 55.0
Reviews of Photo Designer - Write your name with shapes is 3632.0
Reviews of 350 Diy Room Decor Ideas is 27.0
Reviews of FlipaClip - Cartoon animation is 1

Reviews of Manage My Pain Pro is 726.0
Reviews of End Anxiety Pro - Stress, Panic Attack Help is 33.0
Reviews of Acupuncture Assistant is 492.0
Reviews of Number Therapy is 0.0
Reviews of InfantRisk Center HCP is 41.0
Reviews of EMT Tutor NREMT-B Study Guide is 625.0
Reviews of Hospitalist Handbook is 12.0
Reviews of PTA Content Master is 64.0
Reviews of Navi Radiography Pro is 11.0
Reviews of Nursing Care Plan NANDA Tables is 4.0
Reviews of A Manual of Acupuncture is 214.0
Reviews of palmPEDi: Pediatric Tape is 66.0
Reviews of GoodRx Drug Prices and Coupons is 59158.0
Reviews of MyChart is 19473.0
Reviews of FollowMyHealth® is 73118.0
Reviews of CareZone is 27524.0
Reviews of Ovia Pregnancy Tracker & Baby Countdown Calendar is 102858.0
Reviews of Teladoc Member is 2094.0
Reviews of myAir™ for Air10™ by ResMed is 236.0
Reviews of Blood Pressure is 33033.0
Reviews of Pregnancy Week By Week is 78825.0
Reviews of Doctor On Demand is 18674.0
Reviews of BioLife Plasma Services is 250.0
Revi

Reviews of N Launcher: Nougat Theme is 128.0
Reviews of Dino in City-Dinosaur N Police is 10035.0
Reviews of N Launcher Pro - Nougat 7.0 is 2538.0
Reviews of Cops N Robbers - FPS Mini Game is 156862.0
Reviews of Cops N Robbers 2 is 63680.0
Reviews of Hands 'n Guns Simulator is 19727.0
Reviews of Hide N Seek : Mini Game is 4719.0
Reviews of N Launcher - Nougat 7.0 is 32597.0
Reviews of Cook'n Recipe App is 552.0
Reviews of Rope'n'Fly 3 - Dusk Till Dawn is 166033.0
Reviews of Tap N Pay is 253.0
Reviews of N-Com Wizard is 960.0
Reviews of Drag'n'Boom is 133180.0
Reviews of Traps n' Gemstones is 413.0
Reviews of Shop 'n Save is 620.0
Reviews of Dual N-Back - Brain game is 817.0
Reviews of N Files - File Manager & Explorer is 369.0
Reviews of Guns'n'Glory Zombies Premium is 313.0
Reviews of Cops N Robbers is 37789.0
Reviews of Guns'n'Glory Premium is 3570.0
Reviews of Madagascar Surf n' Slides Free is 48929.0
Reviews of Guns'n'Glory is 89947.0
Reviews of Ghosts'n Goblins MOBILE is 466.0
Rev

Reviews of Photo Background Changer 2018 - Blur Background is 34079.0
Reviews of BGCN TV is 4334.0
Reviews of ScorePal is 1439.0
Reviews of Backgammon NJ for Android is 1644.0
Reviews of BG Burger is 1.0
Reviews of JOBS.bg is 1630.0
Reviews of CSCS BG (в български) is 7.0
Reviews of LEGO ® Batman: Beyond Gotham is 10758.0
Reviews of Blood Glucose Tracker is 9612.0
Reviews of Baldur's Gate II is 5442.0
Reviews of Rento - Dice Board Game Online is 205830.0
Reviews of Learn Bulgarian Free is 3049.0
Reviews of N+ News Bulgaria is 2.0
Reviews of Remix Second Hand is 867.0
Reviews of Ultimate Background Eraser is 14453.0
Reviews of OnTrack Diabetes is 6079.0
Reviews of ALL-IN-ONE PACKAGE TRACKING is 167406.0
Reviews of Survival Run with Bear Grylls is 128579.0
Reviews of VIP Backgammon Free : Play Backgammon Online is 389.0
Reviews of Grabo.bg is 8175.0
Reviews of BusyBox Pro is 8114.0
Reviews of GEM™ is 6.0
Reviews of Diabetes Connect is 4027.0
Reviews of B&H Photo Video Pro Audio is 17180.

Reviews of I AM C.T. is 28.0
Reviews of Solar CT PV System Power is 315.0
Reviews of CARDIAC CT TECHNIQUE is 6.0
Reviews of TUTORIAL CT SCAN CARDIAC is 1.0
Reviews of HEAD TRAUMA CT EVALUATION is 18.0
Reviews of Simpli CT is 4.0
Reviews of Remote CT - Smart Remote is 3988.0
Reviews of CT-REMOTE is 19.0
Reviews of CT CONNECT is 2.0
Reviews of CT-ART 4.0 (Chess Tactics 1200-2400 ELO) is 8191.0
Reviews of CT-Stream Player is 84.0
Reviews of CT Pulmonary Angiography is 0.0
Reviews of Results for CT Lottery is 86.0
Reviews of Abdominal CT Sectional Walker is 2.0
Reviews of Interactive CT and MRI Anatomy is 35.0
Reviews of Dine In CT - Food Delivery is 4.0
Reviews of CT Checkout is 1.0
Reviews of CT and XR Dose Calculator is 3.0
Reviews of Chest CT Sectional Walker is 2.0
Reviews of Radiological Anatomy For FRCR1 is 12.0
Reviews of Radiology CT Viewer is 35.0
Reviews of POCKET ATLAS CARDIAC MRI AND CT is 0.0
Reviews of CU: UPI Payments, Chat & Call is 693.0
Reviews of CU of Colorado Mobile B

Reviews of eP Finder is 0.0
Reviews of EP Music is 0.0
Reviews of EP Coding is 0.0
Reviews of EP Kalkulator el. energije is 13.0
Reviews of The Great Wobo Escape Ep.1 is 11.0
Reviews of The Visitor: Ep.1 - Kitty Cat Carnage is 3017.0
Reviews of EP Cook Book is 0.0
Reviews of Dr.Slender Ep 1 Guide (Eng) is 454.0
Reviews of Decay: The Mare - Ep.1 (Trial) is 5291.0
Reviews of EP Search and Find is 20.0
Reviews of What's Your Story?™ ft Charmed is 39661.0
Reviews of EP Gem Hunter is 79.0
Reviews of EP Chain Reaction is 249.0
Reviews of Gate Of Death Ep: 1 is 7.0
Reviews of EP Mobil is 3.0
Reviews of Sonic 4™ Episode I is 8014.0
Reviews of Naruto Shippuden - Watch Free! is 141515.0
Reviews of Lean EQ is 6.0
Reviews of Operate Now: Hospital is 254861.0
Reviews of Multi Surgery ER Emergency Hospital : Doctor Game is 227.0
Reviews of ER Surgery Simulator is 19601.0
Reviews of Dentist Surgery ER Emergency Doctor Hospital Games is 2580.0
Reviews of Eye Transplant : ER Emergency Hospital is 1938.

In [7]:
print(count) 

9660


In [8]:
android_clean=[]
already_added=[]
for app in android:
    name = app[0]
    raw_reviews = app[3]

    try:
        # Clean and convert review count to float
        if 'M' in raw_reviews:
            n_reviews = float(raw_reviews.replace('M', '')) * 1_000_000
           
        elif 'K' in raw_reviews:
            n_reviews = float(raw_reviews.replace('K', '')) * 1_000
            
        else:
            n_reviews = float(raw_reviews.replace(',', ''))  # In case it has commas
        if reviews_max[name]==n_reviews and (name not in already_added):
            android_clean.append(app)
            already_added.append(name)
    except ValueError:
        print(f"Skipping app: {name}, invalid review format: {raw_reviews}")
print(len(android_clean))     

9660


# Remove Non English Apps:
Through this function we check that the name of the apps are in english language.


In [9]:
def is_english(string):
    for c in string:
        if ord(c)>127:
            return False
        return True
print(is_english(''))        

None


Sometimes there is a chance that with app name there is an emojis so handle this problem we only allow only three non ascii values. to reduce loss of data

In [10]:
def is_english(string):
    non_ascii=0
    for c in string:
        if ord(c)>127:
            non_ascii+=1
    if non_ascii>3:
        return False
    else:
        return True
print(is_english('Instagram 😀😀😀'))    

True


In [11]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9615
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

In [12]:
android_final=[]
ios_final=[]
for row in android_english:
    
    price=row[7]
    if price=='0':
        android_final.append(row)

for row in ios_english: 
    price=row[4]
    if price=='0.0':
        ios_final.append(row)
print(len(android_final))
print(len(ios_final))

8864
3222


In [13]:
def freq_table(dataset,index):
    table={}
    total=0
    for app in dataset:
        value=app[index]
        total+=1
        if value in table:
            table[value]+=1
        else:
            table[value]=1
            
    total_percentage={}
    for key in table:
        percentage=(table[key]/total)*100
        total_percentage[key]=percentage
    return total_percentage

def display_table(dataset,index):
    table=freq_table(dataset,index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [14]:
display_table(ios_final,-5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


In [15]:
print(ios_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [16]:
display_table(android_final,1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In [17]:
display_table(android_final, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

# Most Popular Apps by Genre on the App Store
One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

In [23]:
genre_ios=freq_table(ios_final,-5)
for genre in genre_ios: 
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [19]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) # print name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

In [20]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

# Most Popular Apps by Genre on Google Play
For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. 

In [24]:
display_table(android_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


One problem with this data is that is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

In [25]:
categories_android = freq_table(android_final, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [26]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess


If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [27]:
under_100_m = []

for app in android_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386

We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

In [28]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [29]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [30]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

Conclusions
In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
